File: monitoring_activite/objects/conf_anomalie_sonore.php
<?php

class ConfAnomalieSonore{
	
	private $_id_conf_as;
	private $_heure_debut;
	private $_heure_fin;
	private $_seuil;
	private $_duree;
	
	//Constructeur
	public function __construct(array $conf_anomalie_sonore){
		$this->_id_conf_as = (int) $conf_anomalie_sonore['id_conf_as'];
		$this->_heure_debut = $conf_anomalie_sonore['heure_debut'];
		$this->_heure_fin = $conf_anomalie_sonore['heure_fin'];
		$this->_seuil = (int) $conf_anomalie_sonore['seuil'];
		$this->_duree = (int) $conf_anomalie_sonore['duree'];
	}

	//Getters
	public function id_conf_as() { return $this->_id_conf_as; }
	public function heure_debut() { return $this->_heure_debut; }
	public function heure_fin() { return $this->_heure_fin; }
	public function seuil() { return $this->_seuil; }
	public function duree() { return $this->_duree; }

	//Setters
	public function set_heure_debut($heure_debut){ $this->_heure_debut = $heure_debut; }
	public function set_heure_fin($heure_fin){ $this->_heure_fin = $heure_fin; }
	public function set_seuil($seuil){ $this->_seuil = (int) $seuil; }
	public function set_duree($duree){ $this->_duree = (int) $duree; }
	
}
?>

File: Programmes-Qt/BruitLabo/confanomaliesonore.cpp
#include "confanomaliesonore.h"

ConfAnomalieSonore::ConfAnomalieSonore(QTime heure_debut, QTime heure_fin, unsigned int seuil, unsigned int duree, unsigned int id_conf_as, bool debug):
    m_id_conf_as(id_conf_as),
    m_heure_debut(heure_debut),
    m_heure_fin(heure_fin),
    m_seuil(seuil),
    m_duree(duree),
    m_debug(debug){

}

ConfAnomalieSonore::ConfAnomalieSonore(){

}

unsigned int ConfAnomalieSonore::get_id_conf_as() const{
    return m_id_conf_as;
}

QTime ConfAnomalieSonore::get_heure_debut() const{
    return m_heure_debut;
}

QTime ConfAnomalieSonore::get_heure_fin() const{
    return m_heure_fin;
}

unsigned int ConfAnomalieSonore::get_seuil() const{
    return m_seuil;
}

unsigned int ConfAnomalieSonore::get_duree() const{
    return m_duree;
}
